File: content/en/04-llms/llms/_exercises/06_word-games/06_word-games-app.R
library(shiny)
library(ellmer)
library(shinychat)

# Picks a random word for the model to keep secret from a list.
word <- sample(readLines(here::here("data/words.txt")), 1)

system_prompt <- interpolate(
  r"--(
We are playing a word guessing game. The secret word is "{{ word }}".

Never say or otherwise reveal the secret word before the user guesses it.
Give the user an initial clue and then only answer their questions with yes or no.
When they win, use lots of emojis.
)--"
)


# Step 1: Create the chat page UI
ui <- ____

server <- function(input, output, session) {
  # Step 2: Create the chat client with the system prompt
  client <- ____
  # Step 3: Connect the chat server to the chat client
  ____("chat", client)
}

shinyApp(ui, server)

File: content/en/04-llms/llms/_exercises/06_word-games/06_word-games-app.py
import random

import chatlas
from pyhere import here
from shiny import App, ui
from shinychat import Chat, chat_ui

# Picks a random word for the model to keep secret from a list.
word = random.choice(here("data/words.txt").read_text().splitlines())

system_prompt = f"""
We are playing a word guessing game. The secret word is "{word}".

Never say or otherwise reveal the secret word before the user guesses it.
Give the user an initial clue and then only answer their questions with yes or no.
When they win, use lots of emojis.
"""

app_ui = ui.page_fillable(
    # Step 1: Add the chat UI component
    ____
)


def server(input, output, session):
    # Step 2: Initialize the chat client with the system prompt
    client = chatlas.ChatPosit(____=____)
    # Step 3: Connect the chat UI to the chat client
    chat = ____


app = App(app_ui, server)
